### PR TITLE
Assign user

### PR DESCRIPTION
**Pull request by yashhjaggi1998**
Description: Assign user feature allows the client to assign single or multiple users from the Vscode plugin to CxSAST Server.

### CODE REVIEW

File: resultTableWebView.html [hunk 1]
```diff
@@ -24,6 +24,12 @@
     </div>
 
     <script>
+        function getUserPathId(){
```
Comment (umeshwaghode) [inline resultTableWebView.html:27]: Add validations. values can be null,empty, undefined.

File: src/services/WebViews.ts [hunk 2]
```diff
@@ -88,6 +89,18 @@ export class WebViews {
 
 			const resultStates: string[] = await this.httpClient.getRequest(`sast/result-states`);
 			query.resultStates = resultStates;
+
+			let usersResponse = await this.httpClient.getRequest(`auth/AssignableUsers`);
```
Comment (umeshwaghode) [inline src/services/WebViews.ts:93]: Did you  test this API by logging-in with different user that belong to some other team than /CxServer? The assignable users list should vary based on the team (s) to which logged-in user belong.

Comment (umeshwaghode) [reply src/services/WebViews.ts:93]: This list can be fetched only once, not everytime results table is refershed when  High/Medium/low or child nodes on the tree are clicked.

File: src/services/WebViews.ts [hunk 3]
```diff
@@ -185,6 +201,42 @@ export class WebViews {
 			this.log.error(err);
 		}
 	}
+
+	private async assignUser(assignUser: any, rows: any) {
```
Comment (umeshwaghode) [inline src/services/WebViews.ts:205]: Add code comments that describes the way functionality is implemented.

File: src/services/WebViews.ts [hunk 4]
```diff
@@ -185,6 +201,42 @@ export class WebViews {
...
+			var pathId = rows[i];
+			for (let nodeCtr = 0; nodeCtr < nodes.length; nodeCtr++) { 
+				if( pathId == nodes[nodeCtr].Path[0].$.PathId) {
+					const request = {"userAssignment" : assignUser};
+					try {
+						await this.httpClient.patchRequest(`sast/scans/${scanId}/results/${pathId}`, request);
+						nodes[nodeCtr].$.AssignToUser = `${assignUser}`;
+					} catch (err) {
+						if (err.status == 404) {
+							this.log.error('This operation is not supported with CxSAST version in use.');
```
Comment (umeshwaghode) [inline src/services/WebViews.ts:219]: Log 'err' as well.

File: src/services/WebViews.ts [hunk 5]
```diff
@@ -185,6 +201,42 @@ export class WebViews {
...
+		let scanId= this.scanNode.scanId;
+		let nodes = this.queryNode.Result;
+
+		for (var i = 0; i < rows.length; i++) {
+			var pathId = rows[i];
+			for (let nodeCtr = 0; nodeCtr < nodes.length; nodeCtr++) { 
+				if( pathId == nodes[nodeCtr].Path[0].$.PathId) {
+					const request = {"userAssignment" : assignUser};
+					try {
+						await this.httpClient.patchRequest(`sast/scans/${scanId}/results/${pathId}`, request);
```
Comment (umeshwaghode) [inline src/services/WebViews.ts:215]: Confirm that below Tests are performed
1. Assign single vulnerability to a user
2. Repeat for another user for different vulnerability. Refresh results and verify that vulnerabilities are now assigned to respective users.
3. Build assign user to multiple vulnerabilities in one go.
Confirm that these tests are added on the work item.

**Review by umeshwaghode — CHANGES_REQUESTED** [review]
1. I do not see CSS changes to improve look & feel
2.  Update work item with JIRA item link. Update Jira item with final screen captures

File: src/services/WebViews.ts (hunk 5 above)
Comment (yashhjaggi1998) [reply src/services/WebViews.ts]: Done

File: src/services/WebViews.ts (hunk 3 above)
Comment (yashhjaggi1998) [reply src/services/WebViews.ts:213]: Done

**Pull request merged**